04_numeronym/saurabhvyas9/main.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func numeronyms(vals ...string) []string {
+	numeronym := make([]string, 0, len(vals))
+	for _, str := range vals {
```

Comment:
same as labe3: append only if length is not known before:

```go
s := make([]string, len(vals))
for i, str := range vals {
    if len(str) > 3 {
        s[i] =  fmt.Sprintf("%v%d%v", string(str[0]), length-2, string(str[length-1]))
    } else {
        s[i] = str
    }
}

Reply:
Done.